suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,211 @@
+import { A, D, F, pipe } from '@mobily/ts-belt';
+import { memoize, memoizeWithArgs } from 'proxy-memoize';
+
+import { FiatCurrencyCode } from '@suite-common/suite-config';
+import {
+    Account,
+    AccountKey,
+    WalletAccountTransaction,
+    FiatRateKey,
+    FiatRatesStateLegacy,
+    Rate,
+    RateType,
+    TickerId,
+} from '@suite-common/wallet-types';
+import { getFiatRateKeyFromTicker } from '@suite-common/wallet-utils';
+
+import {
+    AccountsRootState,
+    selectAccountByKey,
+    selectDeviceAccounts,
+} from '../accounts/accountsReducer';
+import { TransactionsRootState, selectTransactions } from '../transactions/transactionsReducer';
+import { MAX_AGE, ONE_WEEK_IN_MS } from './fiatRatesConstants';
+import { FiatRatesRootState } from './fiatRatesTypes';
+
+export const selectFiatRatesByFiatRateKey = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+): Rate | undefined => state.wallet.fiat?.[rateType]?.[fiatRateKey];
+
+export const selectIsFiatRateLoading = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+    return currentRate?.isLoading ?? false;
+};
+
+export const selectIsTickerLoading = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+    rateType: RateType = 'current',
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectIsFiatRateLoading(state, fiatRateKey, rateType);
+};
+
+export const selectShouldUpdateFiatRate = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+
+    if (!currentRate) {
+        return true;
+    }
+
+    const { lastSuccessfulFetchTimestamp } = currentRate;
+    const now = Date.now();
```

Comment:
Looks like a code smell to me to have impure selector depending on the current date, but I'm not aware of any direct consequences in this case if it is not memoized. But maybe hook would be a better fit for this. Probably just a nitpick. I will make a note to discuss it on next code improvements meeting.

Reply:
Now I see that this selector is actually used in memoized selector 🤔 I'm really not sure what behaviour to expect from it 🤷‍♂️

Reply:
It was on mobile like this for 8 months and I guess everything works fine... will make a note for code improvements meeting.

Reply:
To me it seems the `now` shall be a parameter of this

Reply:
It is no longer memoized

Reply:
3fc55d303e5c261c77bee41dfa9ed7c0cf140646